crates/router/src/db/merchant_key_store.rs

```diff
@@ -142,7 +142,8 @@ impl MerchantKeyStoreInterface for Store {
 
         #[cfg(feature = "accounts_cache")]
         {
-            let key_store_cache_key = format!("merchant_key_store_{}", merchant_id);
+            let key_store_cache_key =
+                format!("merchant_key_store_{}", merchant_id.get_string_repr());
```

Comment:
these function can move into impl based functions

Reply:
Caching layer will move towards a enum based approach where this will be covered anyways, meanwhile these things can live with merchant_id types impl itself.